lang/syntax.md

```diff
@@ -6,6 +6,7 @@ First, the general structure of programs and functions:
 ```rust
 /// Some opaque type of function names.
 /// The details of this this is represented to not matter.
+#[derive(Hash, PartialEq, Eq, Debug)]
```

Comment:
Not `Clone`?

Reply:
Yeah, `#[derive(Clone)]` is added by specr as of now.
We should probably have some document explaining what specr actually does to the original code ^^'

Reply:
Oh I see.

That'd be the advantage of a `#[specr::name]` attribute -- it could hide all these impls. For Coq we don't want that `Hash` or `Debug` stuff I am pretty sure...

Reply:
So `#[specr::name]` would implement Clone, Copy and maybe GcCompat?

I'm wondering, would there be enums/structs which we don't want to give `#[specr::name]`?

Reply:
`#[specr::name]` is meant as a marker specifically for types that represent 'names' and are used as keys in a `Map`. We don't want to use this for all types.